litmus-portal/frontend/src/models/graphql/prometheus.ts

```diff
@@ -27,10 +27,17 @@ export interface metricsPromResponse {
   tsvs: metricsTimeStampValue[][];
 }
 
+export interface subData {
+  date: number;
+  value: string;
+  subDataName: string;
+}
+
 export interface annotationsPromResponse {
   queryid: string;
   legends: string[];
   tsvs: annotationsTimeStampValue[][];
+  subDataArray: subData[][];
 }
```

Comment:
May you please change `subDataArray`  interfacer to `subData?:
    | Array<{ subDataName: string; value: string; date: number }>
    | undefined;`

Reply:
Its that only @whitetiger1399 this subData array interface is a double dimension one because of the subData for different legends (metrics) its the same reason as for tsv and legends being a double dimension array